server/lib/src/be/mod.rs

```diff
@@ -1704,39 +1710,46 @@ impl<'a> BackendWriteTransaction<'a> {
         // Future idea: Do this in batches of X amount to limit memory
         // consumption.
         let idl = IdList::AllIds;
-        let entries = self.idlayer.get_identry(&idl).map_err(|e| {
-            admin_error!(err = ?e, "get_identry failure");
-            e
+        let entries = self.idlayer.get_identry(&idl).inspect_err(|err| {
+            error!(?err, "get_identry failure");
         })?;
 
         let mut count = 0;
 
+        // This is the longest phase of reindexing, so we have a "progress" display here.
         entries
             .iter()
             .try_for_each(|e| {
-                count += 1;
-                if count % 2500 == 0 {
-                    limmediate_warning!("{}", count);
-                } else if count % 250 == 0 {
-                    limmediate_warning!(".");
+                if immediate {
+                    count += 1;
+                    if count % 2500 == 0 {
+                        eprint!("{}", count);
+                    } else if count % 250 == 0 {
+                        eprint!(".");
+                    }
                 }
+
                 self.entry_index(None, Some(e))
             })
-            .map_err(|e| {
-                admin_error!("reindex failed -> {:?}", e);
-                e
+            .inspect_err(|err| {
+                error!(?err, "reindex failed");
```

Comment:
I'm thinking we need to print a new line before this so the dots don't end on the logline or just look ugly in general

Reply:
I tested this, and surprisingly, we don't. It has a new line properly there.